app/src/main/java/com/example/healthinspector/Fragments/ScanFlow/ProductFinderFragment.java

```diff
@@ -0,0 +1,125 @@
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentManager;
+import androidx.fragment.app.FragmentTransaction;
+import androidx.recyclerview.widget.LinearLayoutManager;
+
+import android.os.Handler;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.example.healthinspector.Adapters.CartItemAdapter;
+import com.example.healthinspector.Adapters.KrogerLocationAdapter;
+import com.example.healthinspector.Constants;
+import com.example.healthinspector.FragmentSwitch;
+import com.example.healthinspector.KrogerLocationCacher;
+import com.example.healthinspector.Models.RecommendedProduct;
+import com.example.healthinspector.Models.ScannedProduct;
+import com.example.healthinspector.R;
+import com.example.healthinspector.databinding.FragmentProductFinderBinding;
+import com.example.healthinspector.databinding.FragmentScanBinding;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.parceler.Parcels;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+
+public class ProductFinderFragment extends Fragment {
+
+    private FragmentProductFinderBinding binding;
+    public static final String TAG = "ProductFinderFragment";
+    private int count;
+    private ArrayList<JSONObject> locations = KrogerLocationCacher.getInstance().getKrogerLocations();
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        binding = FragmentProductFinderBinding.inflate(inflater, container, false);
+        View view = binding.getRoot();
+
+        Bundle bundle = getArguments();
+        RecommendedProduct recommendedProduct = Parcels.unwrap(bundle.getParcelable(Constants.RECOMMENDED_PRODUCT));
+        ScannedProduct scannedProduct = Parcels.unwrap(bundle.getParcelable(Constants.SCANNED_PRODUCT));
+        count = 0;
+
+        if(locations == null || locations.size() == 0){
+            Toast.makeText(requireContext(), getString(R.string.no_nearby_locations), Toast.LENGTH_LONG).show();
+            //navigate user back to product details fragment and finish this fragment
+            FragmentManager fragmentManager = getActivity().getSupportFragmentManager();
+            FragmentTransaction fragmentTransaction =  fragmentManager.beginTransaction();
+            fragmentTransaction.setTransition(FragmentTransaction.TRANSIT_FRAGMENT_OPEN);
+            fragmentManager.popBackStack();
+            fragmentTransaction.commit();
+            return null;
+        }
+        //API request to check all locations for the recommended product's availability
+        for(int i = 0; i < locations.size(); i++){
+            count = i;
```

Comment:
Is 'i' ever used after this point? If not, then just define count in the for loop instead of 'i'

Reply:
I remember getting this error which will prevent my app from running: 

<img width="920" alt="Screen Shot 2022-07-14 at 11 56 38 PM" src="https://user-images.githubusercontent.com/71048942/179169132-3f4860b3-88d4-4ec6-bcc3-acd59800979b.png">

Reply:
I think the error is from using i within the onResponse callback, since the loop will keep running while the onResponse callback gets queued. So I created the count variable as a class attribute and used that